pgmpy/factors/Factor.py

```diff
@@ -780,7 +780,9 @@ def __ne__(self, other):
         return not self.__eq__(other)
 
     def __hash__(self):
-        variables = sorted(self.variables)
+        var_with_hashes = [(var, hash(var)) for var in self.variables]
```

Comment:
@abhi95 You could have used map instead. Also, why you used `(var, hash(var))` not `(hash(var), var)`